node/src/components/storage.rs

```diff
@@ -1196,6 +1219,34 @@ impl StorageInner {
         Ok(outcome)
     }
 
+    /// Puts deploys into storage.
+    pub fn put_deploys<'a, I: IntoIterator<Item = &'a Deploy> + 'a>(
+        &self,
+        deploys: I,
+    ) -> Result<bool, FatalStorageError> {
+        let mut txn = self.env.begin_rw_txn()?;
+        let mut outcome = true;
+        for deploy in deploys {
+            outcome = outcome && txn.put_value(self.deploy_db, deploy.id(), deploy, false)?;
+        }
+        txn.commit()?;
+        Ok(outcome)
+    }
+
+    /// Puts block and its deploys into storage.
+    pub fn put_block_and_deploys(
+        &self,
+        block_and_deploys: &BlockAndDeploys,
+    ) -> Result<bool, FatalStorageError> {
+        let BlockAndDeploys { block, deploys } = block_and_deploys;
+
+        let write_block_result = self.write_block(block)?;
+
+        let write_deploys_result = self.put_deploys(deploys)?;
```

Comment:
I'm wondering now if we shouldn't hold a transaction across both of these operations and not commit the data if writing any of the deploys fails 🤔 What do you think? We don't have good recovery for this kind of errors yet.

Reply:
Good point.  Addressed in [a607d3f](https://github.com/casper-network/casper-node/pull/3010/commits/a607d3fe9cc2a32b520faa06551f0b82b78ef701).